src/Log/LogModel.php

```diff
@@ -0,0 +1,223 @@
+		$this->category    = $category;
+		$this->source      = $source;
+		$this->migrationId = $migrationId;
+		$this->context     = $context;
+		$this->message     = $message;
+		$this->id          = $logId;
+	}
+
+	/**
+	 * Set log type
+	 * If log type is not supported, it will fallback to NOTICE type
+	 *
+	 * @param string $type
+	 */
+	private function setType( $type ) {
+		$this->type = array_key_exists( $type, LogType::getAllTypes() )
+			? $type
+			: LogType::NOTICE;
+	}
+
+	/**
+	 * Set log message
+	 * If not defined, fallback to default value
+	 *
+	 * @param string|null $message
+	 */
+	private function setMessage( $message ) {
+		$this->message = is_null( $message )
+			? esc_html__( 'Something went wrong', 'give' )
+			: $message;
+	}
+
+	/**
+	 * Set log source
+	 * If not defined, fallback to default value
+	 *
+	 * @param string|null $source
+	 */
+	private function setSource( $source ) {
+		$this->source = is_null( $source )
+			? esc_html__( 'Give Core', 'give' )
+			: $source;
+	}
+
+	 /**
+	 * @return int
+	 */
+	public function getId() {
+		return $this->id;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getType() {
+		return $this->type;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getCategory() {
+		return $this->category;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getSource() {
+		return $this->source;
+	}
+
+	/**
+	 * @return string|null
+	 */
+	public function getMigrationId() {
+		return $this->migrationId;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getMessage() {
+		return $this->message;
+	}
+
+	/**
+	 * @return array
+	 */
+	public function getContext() {
+		return $this->context;
+	}
+
+	/**
+	 * Get context data
+	 *
+	 * @param  bool  $jsonEncode
+	 *
+	 * @return string|array
+	 */
+	public function getData( $jsonEncode = false ) {
+		$data = [
+			'message' => $this->getMessage(),
+			'context' => $this->getContext(),
+		];
+
+		if ( $jsonEncode ) {
+			return json_encode( $data );
+		}
+
+		return $data;
+	}
+
+	/**
+	 * @param string $key
+	 * @param mixed $value
+	 */
+	private function addContext( $key, $value ) {
+		if ( is_array( $value ) || is_object( $value ) ) {
+			$value = print_r( $value, true );
+		}
+		$this->context[ $key ] = $value;
+	}
+
+	/**
+	 * @param string $type
+	 * @param array $args
+	 */
+	public function __call( $type, $args ) {
+		list ( $message, $source, $context ) = array_pad( $args, 3, null );
+
+		$this->setType( $type );
+		$this->setMessage( $message );
+		$this->setSource( $source );
+
+		// Set additional context
+		if ( is_array( $context ) ) {
+			foreach ( $context as $key => $value ) {
+				$this->addContext( $key, $value );
+			}
+		}
+
+		$this->save();
+	}
+
+	/**
+	 * Save log record
+	 */
+	public function save() {
+		/**
+		 * var LogRepository $repository
+		 */
+		$repository = give( LogRepository::class );
+
+		$this->getId()
+			? $repository->updateLog( $this )
+			: $repository->insertLog( $this );
```

Comment:
You're not assigning the ID to the model upon `$repository->insertLog( $this )` so subsequent `$model->save()` calls would still insert.

Reply:
:facepalm: 

Resolved: 3ee295a6